docker/ubuntu-14.04-cpu-all-options/Dockerfile

```diff
@@ -0,0 +1,52 @@
+FROM caffe2ai/caffe2:c2v0.8.0.cpu.min.ubuntu14.04
+MAINTAINER Aaron Markham <aaronmarkham@fb.com>
+
+# caffe2 install with cpu support
+# 14.04 uses an older pip so using -U instead of --no-cache-dir
```

Comment:
You're right. Here's what I typically use to get around it:
```Dockerfile
# Don't use deb package because trusty's pip is too old for --no-cache-dir
RUN curl -O https://bootstrap.pypa.io/get-pip.py \
    && python get-pip.py \
    && rm get-pip.py
RUN pip install --no-cache-dir --upgrade pip setuptools wheel
RUN pip install --no-cache-dir future hypothesis pytest
```
If you like having the deb package installed for some reason, you could use this formula instead (though I wouldn't recommend it):
```Dockerfile
RUN apt-get update \
    && apt-get install -y --no-install-recommends \
        python-pip \
    && rm -rf /var/lib/apt/lists/*
# Can't use the --no-cache-dir flag until after pip is upgraded
RUN pip install --upgrade pip
RUN pip install --no-cache-dir --upgrade pip setuptools wheel
RUN pip install --no-cache-dir future hypothesis pytest
```